pkgs/sdk/server/src/Integrations/DataSystemModes.cs

```diff
@@ -0,0 +1,159 @@
+using LaunchDarkly.Sdk.Server.Subsystems;
+
+namespace LaunchDarkly.Sdk.Server.Integrations
+{
+    /// <summary>
+    /// A set of different data system modes which provided pre-configured <see cref="DataSystemBuilder"/>s.
+    /// <para>
+    /// This class is not stable, and not subject to any backwards compatibility guarantees or semantic versioning.
+    /// It is not suitable for production usage. Do not use it. You have been warned.
+    /// </para>
+    /// </summary>
+    internal sealed class DataSystemModes
+    {
+        // This implementation is non-static to allow for easy usage with "Components".
```

Comment:
It would be nice to have this static, but then it doesn't really fit the components pattern at all. This at least makes the data system discoverable at that level. It also isn't a performance concern as this isn't a high frequency operation.